test/test_nn.py

```diff
@@ -15761,9 +15761,10 @@ def test_pooling_bfloat16(self, device):
         self._test_bfloat16_ops(torch.nn.AdaptiveAvgPool2d((3, 5)), device, inp_dims=(8, 4, 16, 16), prec=0.05)
         self._test_bfloat16_ops(torch.nn.AdaptiveAvgPool3d((3, 5, 7)), device, inp_dims=(8, 4, 16, 16, 16), prec=0.05)
 
-    @onlyCUDA
+    @onlyOnCPUAndCUDA
     def test_softmax_bfloat16(self, device):
-        self._test_bfloat16_ops(torch.nn.Softmax(dim=1), device, inp_dims=(16, 32), prec=1e-2)
+        dim = 0 if device == "cpu" else 1
```

Comment:
Does the test fail if we run for one of the dims for one of the devices? Can't we run it for both dims on both devices?

Reply:
To run BF16 in last-dim on CPU, it depends on another PR previously which just has been merged. So I rebase this branch and I think we are good to run both dims on both devices now.